cli/azd/pkg/project/service_target_aks.go

```diff
@@ -0,0 +1,458 @@
+	}
+
+	// Save the name of the image we pushed into the environment with a well known key.
+	t.env.SetServiceProperty(t.config.Name, "IMAGE_NAME", fullTag)
+
+	if err := t.env.Save(); err != nil {
+		return ServiceDeploymentResult{}, fmt.Errorf("saving image name to environment: %w", err)
+	}
+
+	progress <- "Applying k8s manifests"
+	t.kubectl.SetEnv(t.env.Values)
+	deploymentPath := t.config.K8s.DeploymentPath
+	if deploymentPath == "" {
+		deploymentPath = defaultDeploymentPath
+	}
+
+	err = t.kubectl.ApplyFiles(
+		ctx,
+		filepath.Join(t.config.RelativePath, deploymentPath),
+		&kubectl.KubeCliFlags{Namespace: namespace},
+	)
+	if err != nil {
+		return ServiceDeploymentResult{}, fmt.Errorf("failed applying kube manifests: %w", err)
+	}
+
+	deploymentName := t.config.K8s.Deployment.Name
+	if deploymentName == "" {
+		deploymentName = t.config.Name
+	}
+
+	// It is not a requirement for a AZD deploy to contain a deployment object
+	// If we don't find any deployment within the namespace we will continue
+	deployment, err := t.waitForDeployment(ctx, namespace, deploymentName)
+	if err != nil && !errors.Is(err, kubectl.ErrResourceNotFound) {
+		return ServiceDeploymentResult{}, err
+	}
+
+	endpoints, err := t.Endpoints(ctx)
+	if err != nil {
+		return ServiceDeploymentResult{}, err
+	}
+
+	return ServiceDeploymentResult{
+		TargetResourceId: azure.KubernetesServiceRID(
+			t.env.GetSubscriptionId(),
+			t.scope.ResourceGroupName(),
+			t.scope.ResourceName(),
+		),
+		Kind:      AksTarget,
+		Details:   deployment,
+		Endpoints: endpoints,
+	}, nil
+}
+
+// Gets the service endpoints for the AKS service target
+func (t *aksTarget) Endpoints(ctx context.Context) ([]string, error) {
+	namespace := t.getK8sNamespace()
+
+	serviceName := t.config.K8s.Service.Name
+	if serviceName == "" {
+		serviceName = t.config.Name
+	}
+
+	ingressName := t.config.K8s.Service.Name
+	if ingressName == "" {
+		ingressName = t.config.Name
+	}
+
+	// Find endpoints for any matching services
+	// These endpoints would typically be internal cluster accessible endpoints
+	serviceEndpoints, err := t.getServiceEndpoints(ctx, namespace, serviceName)
+	if err != nil && !errors.Is(err, kubectl.ErrResourceNotFound) {
+		return nil, fmt.Errorf("failed retrieving service endpoints, %w", err)
+	}
+
+	// Find endpoints for any matching ingress controllers
+	// These endpoints would typically be publicly accessible endpoints
+	ingressEndpoints, err := t.getIngressEndpoints(ctx, namespace, ingressName)
+	if err != nil && !errors.Is(err, kubectl.ErrResourceNotFound) {
+		return nil, fmt.Errorf("failed retrieving ingress endpoints, %w", err)
+	}
+
+	endpoints := append(serviceEndpoints, ingressEndpoints...)
+
+	return endpoints, nil
+}
+
+// Finds a deployment using the specified deploymentNameFilter string
+// Waits until the deployment rollout is complete and all replicas are accessible
+func (t *aksTarget) waitForDeployment(
```

Comment:
It would be ideal to fail faster for these types of cases... let me see what I can figure out here.

Reply:
Updated to addtionally perform a `kubectl rollout status ...` command check and wait for successful deployment to complete, otherwise provides some level of error detail.

More details on rollout status @ https://kubernetes.io/docs/concepts/workloads/controllers/deployment/